crates/iceberg/src/expr/predicate.rs

```diff
@@ -282,15 +282,18 @@ impl Bind for Predicate {
                             return Ok(BoundPredicate::AlwaysTrue);
                         }
                     }
-                    &PredicateOperator::IsNan | &PredicateOperator::NotNan => {
-                        if !bound_expr.term.field().field_type.is_floating_type() {
-                            return Err(Error::new(
-                                ErrorKind::DataInvalid,
-                                format!(
-                                    "Expecting floating point type, but found {}",
-                                    bound_expr.term.field().field_type
-                                ),
-                            ));
+                    &PredicateOperator::IsNan => {
+                        if bound_expr.term.field().field_type.is_floating_type()
```

Comment:
Sorry, I don't quite the point of this rewriting. Why if `a` is floating or required, `a is nan` is always false?